lib/states/send_money.rb

```diff
@@ -0,0 +1,94 @@
+module States
+  class SendMoney < Base
+    def action
+      puts I18n.t(:send_money_message)
+      select_card_step if account_have_cards?(cards)
+    end
+
+    def next
+      MenuAccount.new(@context)
+    end
+
+    private
+
+    def select_card_step
+      print_cards(cards)
+      selected_card_index = read_input.to_i
+      return unless card_index_valid?(selected_card_index)
+
+      selected_card_index -= 1
+      @sender_card = cards[selected_card_index]
+      receiver_card_number_step
+    end
+
+    def receiver_card_step
+      @receiver_card = card_by_number(@receiver_card_number)
+      return unless card_exists?(@receiver_card, @receiver_card_number)
+
+      amount_step
+    end
+
+    def receiver_card_number_step
+      @receiver_card_number = read_input_with_title(I18n.t(:receiver_card))
+      return unless card_length_valid?(@receiver_card_number)
+
+      receiver_card_step
+    end
+
+    def amount_step
+      @amount = read_input_with_title(I18n.t(:withdraw_amount_message)).to_i
+      return unless amount_valid?(@amount)
+
+      taxes_step
+    end
+
+    def taxes_step
+      @sender_tax_amount = send_tax(@sender_card.type, @amount)
+      @receiver_tax_amount = put_tax(@receiver_card.type, @amount)
+      @sender_balance = @sender_card.balance - @amount - @sender_tax_amount
+      @receiver_balance = @receiver_card.balance + @amount - @receivertax_amount
+      return unless balance_valid?(@sender_balance)
+      return unless receiver_tax_valid?(@receiver_tax_amount, @amount)
+
+      save_balances_step
+    end
+
+    def save_balances_step
+      @sender_card.balance = @sender_balance
+      @receiver_card.balance = @receiver_balance
+      @context.save
+      withdraw_stats(@amount, @sender_card.number, @sender_balance, @sender_tax_amount)
+      put_stats(@amount, @rreceiver_card_number, @receiver_balance, @receiver_tax_amount)
+    end
+
+    def receiver_tax_valid?(receiver_tax_amount, input_amount)
+      return true if receiver_tax_amount < input_amount
+
+      false
+    end
+
+    def card_by_number(card_number)
+      @context.accounts.flat_map(&:cards).detect { |cards| cards.number == card_number }
+    end
+
+    def card_exists?(_card, _card_number)
+      return true unless cards.nil?
+
+      false
+    end
+
+    def card_length_valid?(card_number)
```

Comment:
Do we actually need this method? You use it once, and you can call **card_number.length.between?(15, 17)** inline in code